ntoskrnl/se/token.c

```diff
@@ -1196,13 +1196,28 @@ SepDuplicateToken(
             if (AccessToken->UserAndGroups[GroupsIndex].Attributes == 0 ||
                 (AccessToken->UserAndGroups[GroupsIndex].Attributes & SE_GROUP_ENABLED) == 0)
             {
+                /*
+                 * If this group is an administrators group
+                 * and the token belongs to such group,
+                 * we've to take away TOKEN_HAS_ADMIN_GROUP
+                 * for the fact that's not enabled and as
+                 * such the token no longer belongs to
+                 * this group.
+                 */
+                if (RtlEqualSid(SeAliasAdminsSid,
```

Comment:
Can there be multiple entries for the same SID in the `UserAndGroups` array? If so, you should probably only remove the flag if all of them get removed.

Reply:
No, there cannot be multiple elements of the same SID in the array, at least I have never been aware of nor Windows seems to do any checks regarding this circumstance (and personally, does it make sense to have such entries having the same SID, `SeAliasAdminsSid` in particular?).

Reply:
No, I didn't have any particular use case in mind -- was just checking to make sure, since I noticed it was an implicit assumption that the logic depends on.